src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -207,6 +202,13 @@ public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output,
                         services.AddSingleton<HtmlFactsService, DefaultHtmlFactsService>();
                         services.AddSingleton<WorkspaceDirectoryPathResolver, DefaultWorkspaceDirectoryPathResolver>();
                         services.AddSingleton<RazorComponentSearchEngine, DefaultRazorComponentSearchEngine>();
+                        services.AddSingleton<LanguageServerFeatureOptions, DefaultLanguageServerFeatureOptions>();
+
+                        if (configure != null)
```

Comment:
Moved to end so services in `RazorLanguageServerClient` builder will take precedence over the `Default` services here.

Is there a better way to override? I tried `services.AddSingleton<LanguageServerFeatureOptions, VSLanguageServerFeatureOptions>(_vsLanguageServerFeatureOptions)` but that didn't control precedence.

Reply:
There's a `TryAddSingleton` extension method that will fail to add if a service already exists. Meaning you can leave the configure where it was before but you can change the `DefaultLanguageServerFeatureOptions` addition to be "TryAdd"ed